src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -103,18 +103,12 @@ internal async Task<int> Collect(CancellationToken ct, CommandLineConfiguration
                         Console.WriteLine("--show-child-io must not be specified when attaching to a process");
                         return (int)ReturnCode.ArgumentError;
                     }
-                    if (CommandUtils.ResolveProcessForAttach(processId, name, diagnosticPort, dsrouter, out int resolvedProcessId))
-                    {
-                        processId = resolvedProcessId;
-                    }
-                    else
-                    {
-                        return (int)ReturnCode.ArgumentError;
```

Comment:
Did we lose this error case, when function returns argument error code?

Reply:
You're right, it changes the return error code to `TracingError` because of the catch at https://github.com/dotnet/diagnostics/blob/c0ea514141ba65159bf2ac6f116a8eaf7ff2b97f/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs#L477-L482. Maybe `CommandLineErrorException`, which I'm understanding represents an invalid combination of user's passed-in args, should all result in `ReturnCode.ArgumentError` instead? Otherwise, if we need to differentiate between some `CommandLineErrorException`s throwing ArgumentError and others throwing `TracingError`, that seems like the `CommandLineErrorException` is overloaded

Reply:
> Are there known instances of users invoking dotnet-trace programmatically and checking the exit code?

Not that I am aware of, though that doesn't mean it never happens. My thought would be make sure scripts can distinguish success from failure but don't worry about categorizing different kinds of error with different numeric codes.